org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
> number of _ prefixes depends on the creation order and _ does not imply array.

So what?
How does `[]` make life better?

Let me put it in another way: can you list a test case that would fail unless we use `[]` for arrays?

>There is special alias lookup where are builtin types

There are user-defined types as well, and I expect users would want `_` kind of array names (like the ones for builtin types).

>Basically, [] at the end of type is the only fully correct way to represent arrays.

We can switch from `String` to `SomeTypeReference` class. It would still be a way to represent arrays.